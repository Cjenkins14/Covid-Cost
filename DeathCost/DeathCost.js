import React from 'react';

function Cost(count) {
    let totalCost = parseInt(count.count) * 10000000;
    let string = totalCost.toString();
    let result = string.replace(/(\d)(?=(\d\d\d)+(?!\d))/g, "$1,");

    return (
        <div className='death-cost'>
        <h3>Cumulative deaths: {count.count}</h3>
        <h3>Total Human Cost: ${result}</h3>
        </div>
    )
}

export default Cost;