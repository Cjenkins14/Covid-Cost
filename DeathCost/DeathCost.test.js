import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeathCost from './DeathCost';

describe('Deathcost component', () => {
    const props = {
        count: 1
    }

    it('renders comp by default', () => {
        const wrapper = shallow(<DeathCost/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders the comp given props', () => {
        const wrapper = shallow(<DeathCost {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})