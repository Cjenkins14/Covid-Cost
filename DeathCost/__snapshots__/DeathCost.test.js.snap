// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deathcost component renders comp by default 1`] = `
<div
  className="death-cost"
>
  <h3>
    Cumulative deaths: 
  </h3>
  <h3>
    Total Human Cost: $
    0
  </h3>
</div>
`;

exports[`Deathcost component renders the comp given props 1`] = `
<div
  className="death-cost"
>
  <h3>
    Cumulative deaths: 
    1
  </h3>
  <h3>
    Total Human Cost: $
    10,000,000
  </h3>
</div>
`;
